### PR TITLE
[linting] precommit hooks for linting

### PR DESCRIPTION
**Pull request by tabraiz12**
Add a Precommit hook for linting on the diff

Run the script ./setup-hooks.sh for this to work.

Example Runs only on diffs:
Running ESLint for desktop/core/src/desktop/js/jquery/plugins/jquery.filechooser.js...

/Users/tabraiz/Documents/cloudera/internal-hue/hue/desktop/core/src/desktop/js/jquery/plugins/jquery.filechooser.js
  791:4  error  Parsing error: Unexpected token, expected "," (791:4)

✖ 1 problem (1 error, 0 warnings)

## What changes were proposed in this pull request?

- (Please fill in changes proposed in this fix)

## How was this patch tested?

- (Please explain how this patch was tested. Ex: unit tests, manual tests)
- (If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

**Review by bjornalm — CHANGES_REQUESTED** [review]
Sorry, In know the task stated "pre commit hook" BUT can we put this in a pre-push hook instead? I think the main objective here is to catch more linting issues before the code reaches the CI but without interfering to much with the developers git process in general. 

I often create WIP commits as part of my workflow and it would be annoying if linting would block me there.

File: tools/git-hooks/pre-commit [hunk 1]
```diff
@@ -30,6 +30,13 @@ for FILE in `git diff --name-only` ; do
             ERRORS=true
             echo -e "[ERROR] $FILE: Found pdb.set_trace() debug statements in file.\n"
         fi
+    elif [[ "$FILE" =~ ^.+(js|jsx|ts|tsx)$ ]]; then
```
Comment (JohanAhlen) [inline tools/git-hooks/pre-commit:33]: the `npm run lint` only checks files in certain folders I think it'd be good to restrict this hook to only check for changes in those folders (see package.json)

**Review by JohanAhlen — APPROVED** [review]
Nice!

**Comment by tabraiz12** [comment]
> Sorry, In know the task stated "pre commit hook" BUT can we put this in a pre-push hook instead? I think the main objective here is to catch more linting issues before the code reaches the CI but without interfering to much with the developers git process in general.
> 
> I often create WIP commits as part of my workflow and it would be annoying if linting would block me there.

I have added both the hooks, as it can be a choice for the dev. AS I prefer having it while committing.

File: tools/git-hooks/pre-commit (hunk 1 above)
Comment (tabraiz12) [reply tools/git-hooks/pre-commit:33]: Added Folders check

**Review by JohanAhlen — APPROVED** [review]
Nice!

**Review by bjornalm — APPROVED** [review]
Great work!

**Pull request merged**